android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,354 @@
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+@Composable
+private fun BadgeFiller(
+    badgeType: BadgeType,
+    isStackedBadge: Boolean,
+    badge: BadgeInfo,
+    vector: ImageVector?,
+    isHiddenLayoutFiller: Boolean = false,
+) {
+    Row(
+        modifier = Modifier.padding(
+            horizontal = badgeType.getHorizontalPadding(isStackedBadge),
+            vertical = badgeType.verticalPadding
+        ),
+        horizontalArrangement = Arrangement.spacedBy(constants.spacer)
+    ) {
+        if (isHiddenLayoutFiller) {
+            Box(modifier = Modifier.align(Alignment.CenterVertically).width(constants.iconWidth)) {}
+        } else {
+            vector?.let {
+                Image(
+                    vector,
+                    colorFilter = ColorFilter.tint(badge.contentColor
+                        ?: constants.defaultContentColor),
+                    modifier = Modifier.align(Alignment.CenterVertically)
+                        .width(constants.iconWidth)
+                        .height(constants.iconHeight)
+                )
+            }
+        }
+        badge.text?.let {
+            Text(
+                text = badge.text,
+                color = badge.contentColor ?: constants.defaultContentColor,
+                style = badgeType.textStyle,
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .padding(bottom = 1.dp)
+                    .wrapContentHeight(unbounded = true)
+            )
+        }
+    }
+}
+
+@Composable
+private val BadgeType.badgeHeight: Dp
+    get() = when (this) {
+        Small -> constants.minHeightSmall
+        Medium -> constants.minHeightMedium
+    }
+
+
+@Composable
+private val BadgeType.badgeRounding: Dp
+    get() = when (this) {
+        Small -> constants.cornerRadiusSmall
+        Medium -> constants.cornerRadiusMedium
+    }
+
+
+@Composable
+private fun BadgeType.getHorizontalPadding(isStackedBadge: Boolean): Dp {
+    return when (this) {
+        Small -> constants.horizontalPaddingSmall.plusIf(isStackedBadge,
+            constants.borderWidth)
+        Medium -> constants.horizontalPaddingMedium.plusIf(isStackedBadge,
+            constants.borderWidth)
+    }
+}
+
+@Composable
+private val BadgeType.verticalPadding: Dp
+    get() = when (this) {
+        Small -> 0.dp
+        Medium -> constants.verticalPaddingMedium
+    }
+
+
+@Composable
+private val BadgeType.textStyle: TextStyle
+    get() = when (this) {
+        Small -> constants.textStyleSmall
+        Medium -> constants.textStyleMedium
+    }
+
+enum class BadgeType {
+    Small,
+    Medium,
+}
+
+private fun Dp.plusIf(shouldInclude: Boolean, other: Dp): Dp {
+    return if (shouldInclude) Dp(value = this.value + other.value) else Dp(value = this.value)
```

Comment:
I'm pretty sure `Dp` defines `operator fun plus` so we should be able to write `if (shouldInclude) this + other else this`